### PR TITLE
Add pt-br (Portuguese from Brazil) language file

### PR DESCRIPTION
**Pull request by thiagobonfante**
Just in case you guys to be interested to add support for Portuguese from Brazil

### CODE REVIEW

**Comment by entando-service** [comment]
Can one of the admins verify this patch?

**Comment by joewhite101** [comment]
Hey @thiagobonfante this is awesome. Thanks for the PR. We'll get it reviewed and merged soon

**Comment by joewhite101** [comment]
@sergiofilhowz Can you review this when you get a chance?

File: src/locales/pt.js [hunk 1]
```diff
@@ -0,0 +1,706 @@
...
+    'fileBrowser.createFolder': 'Criar pasta',
+    'fileBrowser.uploadFile': 'Upload',
+    'fileBrowser.uploadFileComplete': 'Upload de arquivo concluído',
+    'fileBrowser.uploadFileError': 'Erro no upload de arquivo',
+    'fileBrowser.downloadFile': 'Download',
+    'fileBrowser.newFolder': 'Nova da Nova Pasta',
+    'fileBrowser.createFolderSuccess': 'Pasta \'{path}\' criada com sucesso',
+    'fileBrowser.createFolderError': 'Ocorreu um erro ao criar a pasta \'{path}\'',
+    'fileBrowser.deleteFolderSuccess': 'Pasta \'{path}\' deletada com sucesso',
+    'fileBrowser.deleteFolderError': 'Ocorreu um erro ao deletar a pasta \'{path}\'',
```
Comment (sergiofilhowz) [inline src/locales/pt.js]: I think we should use the term `remover` instead of `deletar`.

File: src/locales/pt.js [hunk 2]
```diff
@@ -0,0 +1,706 @@
...
+    'fragment.form.edit.plugin': 'Plugin',
+    'fragment.table.widgetType': 'Tipo de Widget',
+    'fragment.table.plugin': 'Plugin',
+    'fragment.table.edit': 'Editar {code}',
+    'fragment.table.details': 'Detalhes de: {code}',
+    'fragment.form.edit.widgetType': 'Tipo de Widget',
+    'fragment.settings': 'Habilitar Edição de Fragmento com interface padrão em branco',
+    'fragment.settings.alert.success': 'As configurações foram atualizadas com sucesso',
+    'fragment.settings.alert.error': 'As configurações não foram atualizadas',
+    'fragment.alert.error.fileExtension': 'Essa extenção não é permitida',
```
Comment (sergiofilhowz) [inline src/locales/pt.js]: Please fix this typo: `extenção` -> `extensão`

File: src/locales/pt.js [hunk 3]
```diff
@@ -0,0 +1,706 @@
...
+    'validateForm.maxLength': 'Deve ter {max} caracteres ou menos',
+    'validateForm.minLength': 'Deve ter {min} caracteres ou mais',
+    'validateForm.number': 'Deve ser um número',
+    'validateForm.minValue': 'Deve ter ao menos {min}',
+    'validateForm.maxValue': 'Deve ter no máximo {max}',
+    'validateForm.email': 'Endereço de email inválido',
+    'validateForm.alphaNumeric': 'Somente caracteres alfanuméricos',
+    'validateForm.widgetCode': '{name} contém caracteres inválidos. Somente caracteres alfanumpericos e sublinhado _ são permitidos',
+    'validateForm.passwordNotMatch': 'Confirm value doesn\'t match with Password value',
+    'validateForm.code': 'Código contém caracteres inválidos. omente caracteres alfanumpericos e sublinhado \'_\' são permitidos.',
```
Comment (sergiofilhowz) [inline src/locales/pt.js]: Please fix these typos: 
* `omente` -> `somente`
* `alfanumpericos` -> `alfanuméricos`

File: src/locales/pt.js [hunk 4]
```diff
@@ -0,0 +1,706 @@
...
+    'validateForm.element.code': 'Código deve ter 3 letras maiúsculas',
+    'validateForm.required': 'Campo obrigatório',
+    'validateForm.maxLength': 'Deve ter {max} caracteres ou menos',
+    'validateForm.minLength': 'Deve ter {min} caracteres ou mais',
+    'validateForm.number': 'Deve ser um número',
+    'validateForm.minValue': 'Deve ter ao menos {min}',
+    'validateForm.maxValue': 'Deve ter no máximo {max}',
+    'validateForm.email': 'Endereço de email inválido',
+    'validateForm.alphaNumeric': 'Somente caracteres alfanuméricos',
+    'validateForm.widgetCode': '{name} contém caracteres inválidos. Somente caracteres alfanumpericos e sublinhado _ são permitidos',
```
Comment (sergiofilhowz) [inline src/locales/pt.js]: Please fix this typo: `alfanuméricos`

File: src/locales/pt.js [hunk 5]
```diff
@@ -0,0 +1,706 @@
...
+    'pageDetails.title': 'Detalhes de Página',
+    'pageDetails.help': 'A seção ÁRVORES DE PÁGINAS permite você gerenciar uma página existente, mostrar ou esconder todas as páginas existentes na árvore ou mostrar todas/esconder todas que pertencem ao nó principal. Você pode também adicionar ou mover uma página, dentre outras ações: preview mobile ou desktop, configurar, editar, ver detalhes, clonar umá página ou deletar. Além disso você pode visualizar o estado (rascunho ou online) se a página estiver na lista de menu.',
+    'pageDetails.emptyContentLink': 'Não há conteúdo com link vinculado a essa página.',
+    'validateForm.elements': 'Insira uma configuração no formato \'key1=value1,key2=value2,key3=value3\'',
+    'validateForm.element': 'Insira uma configuração no formato \'lable1,lable2,lable3\'',
```
Comment (sergiofilhowz) [inline src/locales/pt.js]: Please fix this typo: `lable1,lable2,lable3` -> `label1,label2,label3`

File: src/locales/pt.js [hunk 6]
```diff
@@ -0,0 +1,706 @@
...
+    'pageTree.action.clear': 'Limpar resultados',
+    'pageTree.searchForm.title': 'Buscar por uma página existente',
+    'pageTree.searchForm.code': 'Código de Página',
+    'pageTreePage.help': 'A seção ÁRVORES DE PÁGINAS permite você gerenciar uma página existente, mostrar ou esconder todas as páginas existentes na árvore ou mostrar todas/esconder todas que pertencem ao nó principal. Você pode também adicionar ou mover uma página, dentre outras ações: preview mobile ou desktop, configurar, editar, ver detalhes, clonar umá página ou deletar. Além disso você pode visualizar o estado (rascunho ou online) se a página estiver na lista de menu.',
+    'pageDetails.title': 'Detalhes de Página',
+    'pageDetails.help': 'A seção ÁRVORES DE PÁGINAS permite você gerenciar uma página existente, mostrar ou esconder todas as páginas existentes na árvore ou mostrar todas/esconder todas que pertencem ao nó principal. Você pode também adicionar ou mover uma página, dentre outras ações: preview mobile ou desktop, configurar, editar, ver detalhes, clonar umá página ou deletar. Além disso você pode visualizar o estado (rascunho ou online) se a página estiver na lista de menu.',
```
Comment (sergiofilhowz) [inline src/locales/pt.js]: I think here we should have `Detalhes da Pagina` instead of `ÁRVORES DE PÁGINAS`

File: src/locales/pt.js [hunk 7]
```diff
@@ -0,0 +1,706 @@
...
+    'menu.cms.contentList': 'Lista de Conteúdo',
+    'menu.cms.contentTypes': 'Tipos de Conteúdo',
+    'menu.cms.contentModels': 'Modelos de Conteúdo',
+    'menu.cms.contentSettings': 'Configurações de Conteúdo',
+    'menu.details': 'Detalhes',
+    'menu.groups': 'Grupos',
+    'menu.database': 'Banco de Dados',
+    'menu.widget': 'Widget',
+    'menu.labelsAndLanguages': 'Labels e Idiomas',
+    'menu.uxPattern.addLabels': 'Rótulos de sistema',
```
Comment (sergiofilhowz) [inline src/locales/pt.js]: I think this line should be `Adicionar Rótulos`

File: src/locales/pt.js [hunk 8]
```diff
@@ -0,0 +1,706 @@
...
+    'menu.cms': 'CMS',
+    'menu.cms.contentList': 'Lista de Conteúdo',
+    'menu.cms.contentTypes': 'Tipos de Conteúdo',
+    'menu.cms.contentModels': 'Modelos de Conteúdo',
+    'menu.cms.contentSettings': 'Configurações de Conteúdo',
+    'menu.details': 'Detalhes',
+    'menu.groups': 'Grupos',
+    'menu.database': 'Banco de Dados',
+    'menu.widget': 'Widget',
+    'menu.labelsAndLanguages': 'Labels e Idiomas',
```
Comment (sergiofilhowz) [inline src/locales/pt.js]: I think here we should have `Rótulos e Idiomas`

File: src/locales/pt.js [hunk 9]
```diff
@@ -0,0 +1,706 @@
...
+    'app.milliseconds': 'milissegundos',
+    'app.tableName': 'Nome da Tabela',
+    'app.rows': 'Linhas',
+    'app.monolist': 'Monolist',
+    'app.edit.attribute': 'Editar : atributo - ',
+    'app.edit.Monolist': 'Editar : Monolist',
+    'app.edit.List': 'Editar : Lista',
+    'app.created': 'O {type} {code} foi criado',
+    'app.updated': 'O {type} {code} foi atualizado',
+    'app.deleted': 'O {type} {code} foi deletado',
```
Comment (sergiofilhowz) [inline src/locales/pt.js]: I think we should use the term `removido` instead of `deletado`

File: src/locales/pt.js [hunk 10]
```diff
@@ -0,0 +1,706 @@
...
+    'app.ognl.validation.help1': 'Nessa seção você pode inserir atributos de validação. ',
+    'app.ognl.validation.help2': 'A expressão OGNL deve retornar um valor booleano (true ou false). Na expressão, o objeto pai é o próprio atributo. Você pode acessar os métodos do atributo sem expecificá-lo ou usando o prefixo #atributo.',
+    'app.ognl.validation.help3': 'Você pode o mapa de linguagem do sistema (#langs) e a entidade associada (#entity).',
+    'app.ognl.validation.help4': 'Para elementos de uma lista ou que compõe um atributo você pode acessar o elemento pai (#parent). Além disso, os elementos de uma lista disponibilizam seu índice através da expressão (#index).',
+    'app.ognl.expression': 'Expressão OGNL ',
+    'app.apply.expression': 'Aplique essa expressão apenas para atributos preenchidos',
+    'app.ognl.validation.add.message.help': 'Você pode inserir mensagens de ajuda e de compilação de erros ou escolher um identificador único para anexar elas ao sistema de Labels.',
+    'app.ognl.message': 'Mensagem de erro',
+    'app.help.message.key': 'Identificador da mensagem de ajuda',
+    'app.error.message': 'Error message',
```
Comment (sergiofilhowz) [inline src/locales/pt.js]: I'm not sure about this line, wouldn't it be `Mensagem de erro` ?

File: src/locales/pt.js [hunk 11]
```diff
@@ -0,0 +1,706 @@
...
+    'app.name': 'Nome',
+    'app.group': 'Grupo',
+    'app.code': 'Código',
+    'app.used': 'usado',
+    'app.here': 'aqui',
+    'app.actions': 'Ações',
+    'app.errors': 'Erros',
+    'app.search.returned': 'Sua busca retornou {value} resultados',
+    'app.chooseAnOption': 'Escolha uma opção',
+    'app.add.attribute.code': 'Você deve insireir 3 caracteres maiúsculos',
```
Comment (sergiofilhowz) [inline src/locales/pt.js]: Please fix this typo: `insireir` -> `inserir`

File: src/locales/pt.js [hunk 12]
```diff
@@ -0,0 +1,706 @@
...
+    'app.restore': 'Restaurar',
+    'app.languages': 'Idiomas',
+    'app.systemLabels': 'Rótulos de sistema',
+    'app.cancel': 'Cancelar',
+    'app.mandatory': 'Obrigatório',
+    'app.filterList': 'Pode ser usado como filtros em listas',
+    'app.ognl.validation': 'OGNL - Validação',
+    'app.ognl.validation.help1': 'Nessa seção você pode inserir atributos de validação. ',
+    'app.ognl.validation.help2': 'A expressão OGNL deve retornar um valor booleano (true ou false). Na expressão, o objeto pai é o próprio atributo. Você pode acessar os métodos do atributo sem expecificá-lo ou usando o prefixo #atributo.',
+    'app.ognl.validation.help3': 'Você pode o mapa de linguagem do sistema (#langs) e a entidade associada (#entity).',
```
Comment (sergiofilhowz) [inline src/locales/pt.js]: Please fix this typo: `Você pode o mapa` -> `Você pode usar o mapa`

File: src/locales/pt.js [hunk 13]
```diff
@@ -0,0 +1,706 @@
...
+    'fileBrowser.uploadFileComplete': 'Upload de arquivo concluído',
+    'fileBrowser.uploadFileError': 'Erro no upload de arquivo',
+    'fileBrowser.downloadFile': 'Download',
+    'fileBrowser.newFolder': 'Nova da Nova Pasta',
+    'fileBrowser.createFolderSuccess': 'Pasta \'{path}\' criada com sucesso',
+    'fileBrowser.createFolderError': 'Ocorreu um erro ao criar a pasta \'{path}\'',
+    'fileBrowser.deleteFolderSuccess': 'Pasta \'{path}\' deletada com sucesso',
+    'fileBrowser.deleteFolderError': 'Ocorreu um erro ao deletar a pasta \'{path}\'',
+    'fileBrowser.deleteFileSuccess': 'Arquivo \'{path}\' deletado com sucesso',
+    'fileBrowser.deleteFileError': 'Ocorreu um erro ao deletar o arquivo \'{path}\'',
```
Comment (sergiofilhowz) [inline src/locales/pt.js]: I think we should use the term `remover` instead of `deletar`.

File: src/locales/pt.js [hunk 14]
```diff
@@ -0,0 +1,706 @@
...
+    'fileBrowser.uploadFile': 'Upload',
+    'fileBrowser.uploadFileComplete': 'Upload de arquivo concluído',
+    'fileBrowser.uploadFileError': 'Erro no upload de arquivo',
+    'fileBrowser.downloadFile': 'Download',
+    'fileBrowser.newFolder': 'Nova da Nova Pasta',
+    'fileBrowser.createFolderSuccess': 'Pasta \'{path}\' criada com sucesso',
+    'fileBrowser.createFolderError': 'Ocorreu um erro ao criar a pasta \'{path}\'',
+    'fileBrowser.deleteFolderSuccess': 'Pasta \'{path}\' deletada com sucesso',
+    'fileBrowser.deleteFolderError': 'Ocorreu um erro ao deletar a pasta \'{path}\'',
+    'fileBrowser.deleteFileSuccess': 'Arquivo \'{path}\' deletado com sucesso',
```
Comment (sergiofilhowz) [inline src/locales/pt.js]: I think we should use the term `removido` instead of `deletado`.

File: src/locales/pt.js [hunk 15]
```diff
@@ -0,0 +1,706 @@
...
+    'pageSettings.title': 'Configurações de Página',
+    'pageSettings.help': 'A seção CONFIGURAÇÃO DE PÁGINA permite o administrador especificar que páginas devem ser usadas para um objetivo em específico e permite também definir algumas configurações gerais da página.',
+    'pageSettings.status.published': 'A atualização da página {page} está online!',
+    'pageSettings.status.unpublished': 'A página {page} está offline!',
+    'pageSettings.input.homepage': 'Página Principal',
+    'pageSettings.input.404': 'Página para: 404 - Página não encontrada',
+    'pageSettings.input.500': 'Página para: 500 - Erro genérico',
+    'pageSettings.input.proceed': 'Página para: Faça login para prosseguir',
+    'pageSettings.input.baseURL': 'URL base',
+    'pageSettings.input.baseURL.http': 'Contruído por parâmetros HTTP',
```
Comment (sergiofilhowz) [inline src/locales/pt.js]: Please fix this typo: `Contruído` -> `Construído`

File: src/locales/pt.js [hunk 16]
```diff
@@ -0,0 +1,706 @@
...
+    'pageSettings.input.baseURL.relative': 'Relativo',
+    'pageSettings.input.baseURL.static': 'Estático',
+    'pageSettings.input.appendBaseURL': 'Concaternar nome de contexto à URL base',
```
Comment (sergiofilhowz) [inline src/locales/pt.js]: Please fix this typo: `Concaternar` -> `Concatenar`

File: src/locales/pt.js [hunk 17]
```diff
@@ -0,0 +1,706 @@
...
+    'pageSettings.input.jsession': 'Usar JSESSIONID',
+    'pageSettings.input.languageBroswer': 'Á pagina principal seleciona o seu idioma através do navegador',
+    'pageSettings.input.pageTreeStyle': 'Escolha o estílo da Árvore de Páginas',
```
Comment (sergiofilhowz) [inline src/locales/pt.js]: Please fix this typo: `estílo` -> `estilo`

File: src/locales/pt.js [hunk 18]
```diff
@@ -0,0 +1,706 @@
...
+    'pageSettings.jsession.help': 'É um requisito de segurança. Evite que o JSESSIONID seja inserido no link gerado pelo portal. Deve estar com os cookies ativos.',
+    'pageSettings.appendBaseUrl.help': 'Adiciona o nome de contexto a URL BASE (ex.: /portal/) (ignorado pela BaseURL estática)',
+    'pageSettings.success': 'Configurações de Página foram atualizadas',
+    'dataModel.help': 'A seção MODELOS DE DADOS permite que o administrador adicione novos conteúdos de dados ou edite os existentes. Entando possui um motor de formatação baseado no modelos usados na apresentação dos conteúdos (ou parte disso) usado pelos usuários. Dados de conteúdo são templates Velocity e, assim como os conteúdos, eles são facilmente gerenciados pela interface de administração.',
+    'dataModel.type': 'Tipo',
+    'dataModel.stylesheet': 'Folha de Estilos',
+    'dataModel.model': 'Modelo',
+    'dataModel.deleteDataModelSuccess': 'Modelo de Dados \'{id}\' deletado com sucesso',
+    'dataModel.deleteDataModelError': 'Ocorreu um erro ao deletar o Modelo de Dados \'{id}\'',
```
Comment (sergiofilhowz) [inline src/locales/pt.js]: I think we should use the term `remover` instead of `deletar`.

File: src/locales/pt.js [hunk 19]
```diff
@@ -0,0 +1,706 @@
...
+    'pageSettings.jsession.help': 'É um requisito de segurança. Evite que o JSESSIONID seja inserido no link gerado pelo portal. Deve estar com os cookies ativos.',
+    'pageSettings.appendBaseUrl.help': 'Adiciona o nome de contexto a URL BASE (ex.: /portal/) (ignorado pela BaseURL estática)',
+    'pageSettings.success': 'Configurações de Página foram atualizadas',
+    'dataModel.help': 'A seção MODELOS DE DADOS permite que o administrador adicione novos conteúdos de dados ou edite os existentes. Entando possui um motor de formatação baseado no modelos usados na apresentação dos conteúdos (ou parte disso) usado pelos usuários. Dados de conteúdo são templates Velocity e, assim como os conteúdos, eles são facilmente gerenciados pela interface de administração.',
+    'dataModel.type': 'Tipo',
+    'dataModel.stylesheet': 'Folha de Estilos',
+    'dataModel.model': 'Modelo',
+    'dataModel.deleteDataModelSuccess': 'Modelo de Dados \'{id}\' deletado com sucesso',
```
Comment (sergiofilhowz) [inline src/locales/pt.js]: I think we should use the term `removido` instead of `deletado`.

File: src/locales/pt.js [hunk 20]
```diff
@@ -0,0 +1,706 @@
...
+    'database.datasourceDetails': 'Detalhes do Datasource',
+    'labelsAndLanguages.help': 'A seção LABEL & IDIOMAS permite que você gerencie labels e idiomas do sistema. Entando oferece suporte a multiplos idiomas, o administrador pode adiionar um nomo idioma. O primeiro idioma a ser configurado no sistema será o idioma padrão. Labels são acessórios da Entando, principalmente usados para mostrar informações textuais estáticas na aplicação e não há objetos em particular.',
+    'label.searchForm.code': 'Busca por código',
+    'label.searchForm.title': 'Procucar label existente',
+    'language.selectLanguage': 'Selecione o idioma',
+    'language.active.listEmpty': 'Não há IDIOMAS ATIVOS disponíveis',
+    'label.detail.title': 'Labels',
+    'labels.code.placeholder': 'código',
+    'labels.default.language': 'Idioma padrão',
+    'modal.confirm.delete': 'Você deseja realmente deletar o código {code}?',
```
Comment (sergiofilhowz) [inline src/locales/pt.js]: I think we should use the term `remover` instead of `deletar`.

File: src/locales/pt.js [hunk 21]
```diff
@@ -0,0 +1,706 @@
...
+    'reference.noReferencejacmsResourceManager': 'Não existem recursos referenciados',
+    'reference.noReferenceDataObjectManager': 'Não existem objetos de dados referenciados',
+    'reference.noReferencejpcollaborationIdeaManager': 'Não há idéia referenciada',
+    'references.referencedPages': 'Páginas referenciadas',
+    'reloadConfiguration.title': 'Recarregar configuração',
+    'reloadConfiguration.help': 'A seção RECARREGAR CONFIGURAÇÃO permite recarregar as configurações do sistema. Essa operação precisa ser executada após modificar alguns parâmetros.',
+    'reloadConfiguration.reload.title': 'Recarregar a configuração',
+    'reloadConfiguration.reload.confirm': 'Você tem certeza que deseja recarregar a configuração?',
+    'reloadConfiguration.confirm.success': 'A configuração foi recarregada com sucesso.',
+    'reloadConfiguration.confirm.error': 'Ocorreu algo de errado enquanto recarregavamos a configuração. Tente novamente em alguns minutos.',
```
Comment (sergiofilhowz) [inline src/locales/pt.js]: Please fix this typo: `recarregavamos` -> `recarregávamos`

File: src/locales/pt.js [hunk 22]
```diff
@@ -0,0 +1,706 @@
...
+    'activityStream.modifyPage': 'modificou uma página',
+    'activityStream.like': 'Like',
+    'activityStream.writeComment': 'Escreva um comentário',
+    'reference.text': 'Recarregue as referências para esses tipos de entidade. Tipos a serem recarregados: {count}. ',
+    'reference.reload': 'Clique aqui {link} para recarregar todos os tipos de dados.',
+    'digitalExchange.disabled.title': 'Digital Exchange está desabilitado.',
+    'digitalExchange.disabled.help': 'Você pode habilitar através da variável de ambiente DIGITAL_EXCHANGE_UI_ENABLED (altere para true) que está localizada n arquivo .env.',
+    'digitalExchange.menuButton.title': 'Digital Exchange',
+    'digitalExchange.component.list.title': 'Digital Exchange',
+    'digitalExchange.component.help': 'Entando Digital Exchange é um marketplace onde usuário podem instalar diferentes Widgets, Componentes, Fragmentos, Modelos de Página, Modelos de Conteúdo e Tipos de Conteúdo.',
```
Comment (sergiofilhowz) [inline src/locales/pt.js]: Please fix this typo: `usuário` -> `os usuários`

File: src/locales/pt.js (hunk 1 above)
Comment (thiagobonfante) [reply src/locales/pt.js]: Totally agreed

**Comment by thiagobonfante** [comment]
> Hey @thiagobonfante this is awesome. Thanks for the PR. We'll get it reviewed and merged soon

Hey @joewhite101 anytime!

> Hey @thiagobonfante, thank you very much for your contribution, I've made some notes on it with some minor typo and changing the term `deletar` to `remover`, not a big deal, but I think it's more suitable.

Hey @sergiofilhowz anytime man. Thank you for the quick reply and review. I think it's all good now!

**Comment by pietrangelo** [comment]
retest this please

**Comment by pietrangelo** [comment]
retest this please

**Comment by pietrangelo** [comment]
retest this please

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by pietrangelo** [comment]
add to whitelist

**Comment by pietrangelo** [comment]
retest this please

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by joewhite101** [comment]
retest this please

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
All is well

**Comment by gidesan** [comment]
Thanks a lot, @thiagobonfante :)

**Comment by thiagobonfante** [comment]
@gidesan anytime :)

**Pull request merged**